base/MAPL_TimeMethods.F90

```diff
@@ -85,52 +93,74 @@ function define_time_variable(this,rc) result(v)
     integer :: status
     character(len=ESMF_MAXSTR) :: startTime,timeUnits
     type(ESMF_Time) :: currTime
-    integer :: i1,i2,i3,i123,ipos1,ipos2,isc,imn,ihr,ifreq
+    integer :: i1,i2,i3,ipos1,ipos2,isc,imn,ihr
+    logical :: integer_representable
+    integer :: begin_date, begin_time, time_increment, packed_hms
 
-    call ESMF_CLockGet(this%clock,currTime=currTime,rc=status)
+    _ASSERT(this%frequency/=TimeData_uninit_int,"TimeData object was not created before use")
+
+    call ESMF_ClockGet(this%clock,currTime=currTime,rc=status)
     _VERIFY(status)
     currTime = currTime - this%offset
     call ESMF_TimeGet(currTime,timeString=StartTime,rc=status)
     _VERIFY(status)
     timeUnits = trim(this%funits)//" since "//startTime( 1: 10)//" "//startTime(12:19)
-    v = Variable(type=PFIO_REAL32,dimensions='time')
-    call v%add_attribute('long_name','time')
-    call v%add_attribute('units',trim(timeUnits))
   
     ipos1=index(startTime,"-")
     ipos2=index(startTime,"-",back=.true.)
     read(startTime(1:ipos1-1),'(i4)')i1
     read(startTime(ipos1+1:ipos2-1),'(i2)')i2
     read(startTime(ipos2+1:10),'(i2)')i3
-    i123=10000*i1+100*i2+i3
-    call v%add_attribute('begin_date',i123)
+    begin_date=10000*i1+100*i2+i3
 
     ipos1=index(startTime,":")
     ipos2=index(startTime,":",back=.true.)
     read(startTime(12:ipos1-1),'(i2)')i1
     read(startTime(ipos1+1:ipos2-1),'(i2)')i2
     read(startTime(ipos2+1:19),'(i2)')i3
-    i123=10000*i1+100*i2+i3
-    call v%add_attribute('begin_time',i123)
-
+    begin_time=10000*i1+100*i2+i3
+
+    isc=mod(this%frequency,60)
+    i2=this%frequency-isc
+    i2=i2/60
+    imn=mod(i2,60)
+    i2=i2-imn
+    ihr=i2/60
+    packed_hms=10000*ihr+100*imn+isc 
     select case(trim(this%funits))
+    case('seconds')
+       integer_representable = .true.
+       time_increment = packed_hms
     case('minutes')
-       isc=mod(this%frequency,60)
-       i2=this%frequency-isc
-       i2=i2/60
-       imn=mod(i2,60)
-       i2=i2-imn
-       ihr=i2/60
-       ifreq=10000*ihr+100*imn+isc 
+       integer_representable = mod(this%frequency,60)==0
+       time_increment = packed_hms
+    case('hours')
+       integer_representable = mod(this%frequency,3600) == 0
+       time_increment = packed_hms
     case('days')
-       ifreq = this%frequency/86400
+       integer_representable = mod(this%frequency,86400) == 0
+       time_increment = this%frequency/86400
     case default
        _ASSERT(.false., 'Not supported yet')
     end select
-    call v%add_attribute('time_increment',ifreq)
+
+    if (this%integer_time) then
+        _ASSERT(integer_representable,"time interval requested not integer representable")
+    end if
```

Comment:
Would it be clearer to eliminate `integer_representable` and just put a tailored ASSERT inside each CASE statement?